src/main/java/it/gov/pagopa/payhub/activities/activity/treasury/TreasuryOpiIngestionActivityImpl.java

```diff
@@ -0,0 +1,80 @@
+package it.gov.pagopa.payhub.activities.activity.treasury;
+
+import it.gov.pagopa.payhub.activities.dao.IngestionFlowFileDao;
+import it.gov.pagopa.payhub.activities.dto.IngestionFlowFileDTO;
+import it.gov.pagopa.payhub.activities.dto.treasury.IufIuvDTO;
+import it.gov.pagopa.payhub.activities.dto.treasury.TreasuryIngestionResultDTO;
+import it.gov.pagopa.payhub.activities.exception.IngestionFlowFileNotFoundException;
+
+import it.gov.pagopa.payhub.activities.service.ingestionflow.IngestionFlowFileRetrieverService;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.*;
+
+/**
+ * Interface for the TreasuryOpiIngestionActivity.
+ * Defines methods for processing files based on an IngestionFlow ID.
+ */
+@Slf4j
+@Lazy
+@Component
+public class TreasuryOpiIngestionActivityImpl implements TreasuryOpiIngestionActivity {
+  private final String ingestionflowFileType;
+  private final IngestionFlowFileDao ingestionFlowFileDao;
+  private final IngestionFlowFileRetrieverService ingestionFlowFileRetrieverService;
+
+
+  public TreasuryOpiIngestionActivityImpl(@Value("${ingestion-flow-file-type:O}") String ingestionflowFileType,
```

Comment:
let's start using an Enum for this field
it will not be a variable configurable through properties